.circleci/config.yml

```diff
@@ -1,72 +1,48 @@
   python: circleci/python@1.5.0
-
-# Define a job to be invoked later in a workflow.
-# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
 jobs:
-  build-docs: # This is the name of the job, feel free to change it to better match what you're trying to do!
-    # These next lines defines a Docker executors: https://circleci.com/docs/2.0/executor-types/
-    # You can specify an image from Dockerhub or use one of the convenience images from CircleCI's Developer Hub
-    # A list of available CircleCI Docker convenience images are available here: https://circleci.com/developer/images/image/cimg/python
-    # The executor is the environment in which the steps below will be executed - below will use a python 3.10.2 container
-    # Change the version below to your required version of python
+  build-docs:
     docker:
       - image: cimg/python:3.10.2
-    # Checkout the code as the first step. This is a dedicated CircleCI step.
-    # The python orb's install-packages step will install the dependencies from a Pipfile via Pipenv by default.
-    # Here we're making sure we use just use the system-wide pip. By default it uses the project root's requirements.txt.
-    # Then run your tests!
-    # CircleCI will report the results back to your VCS provider.
     steps:
-      - checkout
+      - checkout:
+          path: napari
+      - run:
+          name: Clone docs repo into a subdirectory
+          command: git clone git@github.com:napari/docs.git docs
       - run:
           name: Install qt libs + xvfb
           command: sudo apt-get update && sudo apt-get install -y xvfb libegl1 libdbus-1-3 libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xinput0 libxcb-xfixes0 x11-utils
-
       - run:
-          name: Install python dependencies
-          # app-dir: ~/project/package-directory/  # If you're requirements.txt isn't in the root directory.
+          name: Setup virtual environment
           command: |
             python -m venv venv
             . venv/bin/activate
             python -m pip install --upgrade pip
-      - run:
-          name: Clone docs repo
-          command: git clone git@github.com:napari/docs.git napari-docs
+
       - run:
           name: Install napari-dev
           command: |
             . venv/bin/activate
-            python -m pip install -e ".[pyside,dev]" -c resources/constraints/constraints_py3.10_docs.txt
+            python -m pip install -e "napari/[pyside,dev]"
+          environment:
+            PIP_CONSTRAINT: napari/resources/constraints/constraints_py3.10_docs.txt
       - run:
           name: Build docs
           command: |
             . venv/bin/activate
-            cd napari-docs
-            xvfb-run --auto-servernum make docs GALLERY_PATH=../../examples/
+            cd docs
+            xvfb-run --auto-servernum make docs
```

Comment:
Did you check to see if this runs? I remember having trouble with the paths where the repos were cloned on CircleCI, which is why I had to explicitly point to the GALLERY_PATH

Reply:
yes. It works because I have change the checkout path.